pex/pip.py

```diff
@@ -297,24 +299,55 @@ def _spawn_pip_isolated(
             from pex.pex import PEX
 
             pip = PEX(pex=self._pip_pex_path, interpreter=interpreter)
-            return pip.cmdline(command), pip.run(args=command, env=env, blocking=False)
+            return pip.cmdline(command), pip.run(
+                args=command, env=env, blocking=False, **popen_kwargs
+            )
 
     def _spawn_pip_isolated_job(
         self,
         args,  # type: Iterable[str]
         package_index_configuration=None,  # type: Optional[PackageIndexConfiguration]
         cache=None,  # type: Optional[str]
         interpreter=None,  # type: Optional[PythonInterpreter]
+        pip_verbosity=0,  # type: int
+        **popen_kwargs  # type: Any
     ):
         # type: (...) -> Job
         command, process = self._spawn_pip_isolated(
             args,
             package_index_configuration=package_index_configuration,
             cache=cache,
             interpreter=interpreter,
+            pip_verbosity=pip_verbosity,
+            **popen_kwargs
         )
         return Job(command=command, process=process)
 
+    def _iter_platform_args(
+        self,
+        platform,  # type: str
+        impl,  # type: str
+        version,  # type: str
+        abi,  # type: str
+        manylinux=None,  # type: Optional[str]
+    ):
+        # type: (...) -> Iterator[str]
+        if manylinux and platform.startswith("linux"):
+            yield "--platform"
+            yield platform.replace("linux", manylinux, 1)
+
+        yield "--platform"
+        yield platform
```

Comment:
Should that second part be in an else clause? Otherwise we'd get `--platform` twice.

Reply:
If that's intentional, a brief comment acknowledging this is intentional would help.

Reply:
Sure. It won't be brief though. This was the reason for the 1st year of running with patched Pip when Pex 1st switched to Pip to support Twitter's use case. It'll take a few paragraphs but be a bit more direct than the 5 tests that fail when you remove it.